subsys/bluetooth/shell/bt.c

```diff
@@ -193,65 +193,84 @@ static bool data_cb(struct bt_data *data, void *user_data)
 	}
 }
 
-static void print_data_set(uint8_t type, uint8_t set_value_len,
-			   const uint8_t *scan_data, uint8_t scan_data_len)
+static void print_data_hex(const uint8_t *data, uint8_t len)
 {
-	uint8_t min_value_len = MIN(set_value_len, scan_data_len);
+	shell_fprintf(ctx_shell, SHELL_INFO, "0x");
+	for (int8_t i = len-1; i >= 0; i--) {
+		shell_fprintf(ctx_shell, SHELL_INFO, "%02x", data[i]);
+	}
+}
 
-	shell_fprintf(ctx_shell, SHELL_INFO, "%*sType 0x%02x: ",
-		      strlen(scan_response_label), "",
-		      type);
-	for (uint8_t i = 0U; i < (scan_data_len / min_value_len); i++) {
-		if (i > 0L) {
-			shell_fprintf(ctx_shell, SHELL_INFO, ", ");
-		}
+static void print_data_set(uint8_t set_value_len, bool single_value,
+			   const uint8_t *scan_data, uint8_t scan_data_len)
+{
+	uint8_t idx = 0;
 
-		shell_fprintf(ctx_shell, SHELL_INFO, "0x");
-		for (uint8_t j = 0U; j < min_value_len; j++) {
-			shell_fprintf(ctx_shell, SHELL_INFO, "%02x",
-				      scan_data[i * min_value_len + j]);
+	do {
+		if (idx > 0) {
+			shell_fprintf(ctx_shell, SHELL_INFO, ",");
```

Comment:
```suggestion
			shell_fprintf(ctx_shell, SHELL_INFO, ", ");
```

Suggest to add a space after the comma to make it easier to read

Reply:
I removed the space after the comma to make it (slightly) cleaner when using a string tokenizer (avoiding the extra step of removing spaces).  If you feel it should include the space for better visibility in the terminal, I can go with that

Reply:
I do believe that `0x12, 0x13, 0x14` is easier to read than `0x12,0x13,0x14` and this is intended for humans after all :D